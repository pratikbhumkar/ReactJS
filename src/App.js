import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

var variab=" From a variable!!"

class App extends Component {
  render() {
    return (
      <div className="App">
      
        {/* <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React by Pratik Bhumkar "Hello world"
          </a>
        </header> */}
        Hey there by Pratik Bhumkar {variab}
        <p><Welcome text='whoas'></Welcome></p>
        
      </div>
    );
  }
}

class Welcome extends Component{
   
  render(){
    
    return(
      <div>
      <p>{this.props.text}</p>
      <p> welcome to pratik's React Component </p>
      </div>);
  };
}
export default App;
