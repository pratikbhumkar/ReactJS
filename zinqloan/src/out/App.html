<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: App</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: App</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>App<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">This is the login component of the application.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="App"><span class="type-signature"></span>new App<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="handleLogins"><span class="type-signature"></span>handleLogins<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    This method does the verification based on the username and password inputs. It verifies and redirects user to the welcome page. If input is wrong then shows error message.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line46">line 46</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="handlePasswordChange"><span class="type-signature"></span>handlePasswordChange<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    This method handles changes in the password
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line38">line 38</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="handleRegister"><span class="type-signature"></span>handleRegister<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    This method redirects user to the create window, once clicked.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line98">line 98</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="handleUsernameChange"><span class="type-signature"></span>handleUsernameChange<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    This method handles changes in the username
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line31">line 31</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Rendering the Login page.
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="App.js.html">App.js</a>, <a href="App.js.html#line108">line 108</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Wed Jan 16 2019 03:20:50 GMT+1100 (Australian Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>