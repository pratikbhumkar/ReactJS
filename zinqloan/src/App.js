import React, { Component } from 'react';
import './App.css';
import ReactDOM from 'react-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import Create from './Create.js'
import { BrowserRouter as Router } from "react-router-dom";
import Welcome from './welcome';
import User from './Models/UserModel'
import { GoogleLogin } from 'react-google-login';
// import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props'
import FacebookLogin from 'react-facebook-login';
/**
 * This is the login component of the application.
 */
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {username: '',
                  username_error:'',
                  password: '',
                  password_error:'',
                  error:[]
                }
    this.handleUsernameChange = this.handleUsernameChange.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.responseForGoogle = this.responseForGoogle.bind(this);
    this.responseForFacebook = this.responseForFacebook.bind(this);
    
  }
/**
 * This method handles changes in the username 
 * @param {*} event 
 */
  handleUsernameChange(event) {
    this.setState({username: event.target.value});
  }
/**
 * This method handles changes in the password 
 * @param {*} event 
 */
  handlePasswordChange(event) {
    this.setState({password: event.target.value});
  }

/**
 * This method does the verification based on the username and password inputs. It verifies and redirects user to the welcome page. If input is wrong then shows error message.
 * @param {*} event 
 */
  handleLogins(event) {
    //To store the errors
    const errors={}

    //Regex to check email is valid
    var re =  new RegExp(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
    var emailVer=re.test(String(this.state.username).toLowerCase())
    //Regex to store check password is valid
    var strongRegex = new RegExp(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/);
    var PassVer=strongRegex.test(String(this.state.password))
    //Flag to determine that the contents are valid.
    var flagInvalid=false

    //Setting the error message if email id is invalid.
    if(emailVer === false){
      errors.username_error="Email address is invalid, Re-enter."
      flagInvalid=true
    }
    //Setting the error message if password is invalid.
    if(PassVer === false){
      errors.password_error="Password is invalid, Re-enter."
      flagInvalid=true
    }
    // If everything is valid, connect to database and check for correctness.
    if(flagInvalid==false){
        fetch(`http://localhost:5000/user/get?email=${this.state.username}&password=${this.state.password}`)
        .then((response) => { 
          return response.json() 
        }).then((response) => {
        var user=new User()
        user.setUserFirstName(response.data[0].User_FirstName)
        user.setUserLastName(response.data[0].User_LastName)
          
            ReactDOM.render((
              <Router>
                <Welcome UserObj={user}/>
              </Router>
            ), document.getElementById('root'))
        })
        .catch(()=> alert("Login Failed please re-enter!"))
            }
      else{
        this.setState({
          ...this.state,
          ...errors
        })
      }
    }
    responseForFacebook(facebookUser){
      var name=(facebookUser.name)
      var user=new User()
      var firstname= name.split(' ')[0]
      var lastname= name.split(' ')[1]
      user.setUserFirstName(firstname)
      user.setUserLastName(lastname)
      
      ReactDOM.render((
        <Router>
          <Welcome UserObj={user}/>
        </Router>
      ), document.getElementById('root'))

    }
    responseForGoogle(googleUser){
      
      var user=new User()
      user.setUserFirstName(googleUser.w3.ofa)
      user.setUserLastName(googleUser.w3.wea)
      ReactDOM.render((
        <Router>
          <Welcome UserObj={user}/>
        </Router>
      ), document.getElementById('root'))
    }
/**
 * This method redirects user to the create window, once clicked.
 * @param {*} event 
 */
  handleRegister(event) {
    ReactDOM.render((
      <Router>
        <Create />
      </Router>
    ), document.getElementById('root'))
  }
  /**
   * Rendering the Login page.
   */
  render() {
    const responseGoogleFail = (response) => {
      console.log(response)
    }
    const responseGoogle = (responseFromGoogle) => {
      this.responseForGoogle(responseFromGoogle)
    }
    // const responseFacebook = (responseFromFacebook) => {
    //   alert('called')
    //   console.log(responseFromFacebook.name)
    //   this.responseForFacebook(responseFromFacebook)
    // }
    const content=<div>
    <MuiThemeProvider>
      <div className="login" >
       <form style={{textAlign:"center"}}>
       <AppBar title="Zinq"/>
            <h1>Login</h1>
              <TextField
              autoFocus
              autoComplete
              hintText="username"
              value={this.state.username}
              errorText={this.state.username_error}
              id="useremail"
              error={this.state.error.usn}
              type="email"
              floatingLabelText="Enter username here"
              style={{ alignItems: 'center'}}
              onChange = {this.handleUsernameChange}
              />
            <br/>
            <TextField
              id="userpwd"
              value={this.state.password}
              hintText="password"
              floatingLabelText="Enter password here"
              errorText={this.state.password_error}
              type= "password"
              error={this.state.error.pwd}
              style={{ alignItems: 'center'}}
              onChange = {this.handlePasswordChange}
              />
            <br/><br/>
            <RaisedButton id="btnLogin" label="Login" primary={true} style={{margin: 15,minWidth: 150}} onClick={(event) => { this.handleLogins() }}/>
            <RaisedButton id="btnCreate" label="Create Account" primary={true} style={{margin: 15 ,minWidth: 150}} onClick={(event) => this.handleRegister()}/>   
            <br/>
            <br/>
            <GoogleLogin
              clientId="228016333193-44664g1hnhq023mo7pv1i4itmeduu1df.apps.googleusercontent.com"
              buttonText="Login"
              onSuccess={responseGoogle}
              onFailure={responseGoogleFail}
            />
            <br/>
            <br/>
            <br/>
            <br/>
             <FacebookLogin
                appId="2150953554964623"
                autoLoad={true}
                fields="name,email"
                // onClick={componentClicked}
                callback={this.responseForFacebook} />
            
        </form> 
        
      </div>
    </MuiThemeProvider> 
  </div>
 
  
 
    return(content);
  }
}

export default App;
