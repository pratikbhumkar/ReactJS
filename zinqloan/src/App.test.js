import React from 'react';
import App from './App';
import {Enzyme,shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure,wrapper } from 'enzyme';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  const editor =shallow(<App/>) 
  expect(editor.find('AppBar').length).toEqual(1)
  expect(editor.find('RaisedButton').length).toEqual(2)
  expect(editor.find('TextField').length).toEqual(2)
  expect(editor.find('h1').length).toEqual(1)
});


it('renders correctly', () => {
  const tree = renderer
    .create(<App/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});