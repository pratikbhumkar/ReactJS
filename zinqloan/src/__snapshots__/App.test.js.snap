// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="login"
  >
    <form
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#00bcd4",
            "borderRadius": "0px",
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "flex",
            "fontFamily": "Roboto, sans-serif",
            "muiPrepared": true,
            "paddingLeft": 24,
            "paddingRight": 24,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
            "zIndex": 1100,
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOut={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 0,
              "fontWeight": "inherit",
              "height": 48,
              "margin": 0,
              "marginLeft": -16,
              "marginRight": 8,
              "marginTop": 8,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "padding": 12,
              "position": "relative",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": 48,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "height": 24,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </div>
        </button>
        <h1
          onClick={[Function]}
          style={
            Object {
              "boxFlex": 1,
              "color": "#ffffff",
              "flex": "1",
              "fontSize": 24,
              "fontWeight": 400,
              "height": 64,
              "letterSpacing": 0,
              "lineHeight": "64px",
              "margin": 0,
              "muiPrepared": true,
              "overflow": "hidden",
              "paddingTop": 0,
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          Zinq
        </h1>
      </div>
      <h1>
        Login
      </h1>
      <div
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 256,
          }
        }
      >
        <label
          htmlFor="useremail"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Enter username here
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          username
        </div>
        <input
          autoComplete={true}
          autoFocus={true}
          disabled={false}
          id="useremail"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="email"
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <br />
      <div
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "height": 72,
            "lineHeight": "24px",
            "muiPrepared": true,
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": 256,
          }
        }
      >
        <label
          htmlFor="userpwd"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "lineHeight": "22px",
              "muiPrepared": true,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 38,
              "transform": "scale(1) translate(0, 0)",
              "transformOrigin": "left top",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
        >
          Enter password here
        </label>
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 0,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          password
        </div>
        <input
          disabled={false}
          id="userpwd"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitOpacity": 1,
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "height": "100%",
              "marginTop": 14,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="password"
        />
        <div>
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderColor": "#e0e0e0",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <hr
            aria-hidden="true"
            style={
              Object {
                "borderBottomStyle": "solid",
                "borderBottomWidth": 2,
                "borderColor": "#00bcd4",
                "borderLeft": "none",
                "borderRight": "none",
                "borderTop": "none",
                "bottom": 8,
                "boxSizing": "content-box",
                "margin": 0,
                "muiPrepared": true,
                "position": "absolute",
                "transform": "scaleX(0)",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <br />
      <br />
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 15,
            "minWidth": 150,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          id="btnLogin"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Login
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 15,
            "minWidth": 150,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          id="btnCreate"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Create Account
              </span>
            </div>
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;
